mailbox/plugin/quota-mailing/src/test/java/org/apache/james/eventsourcing/EventSourcingSystemTest.java

```diff
@@ -0,0 +1,194 @@
+            eventStore);
+
+        eventSourcingSystem.dispatch(new MyCommand(PAYLOAD_1));
+
+        assertThat(subscriber.getData()).containsExactly(PAYLOAD_1);
+    }
+
+    @Test
+    default void dispatchShouldApplyCommandHandlerThenStoreGeneratedEvents(EventStore eventStore) {
+        DataCollectorSubscriber subscriber = new DataCollectorSubscriber();
+        EventSourcingSystem eventSourcingSystem = new EventSourcingSystem(
+            ImmutableSet.of(simpleDispatcher(eventStore)),
+            ImmutableSet.of(subscriber),
+            eventStore);
+
+        eventSourcingSystem.dispatch(new MyCommand(PAYLOAD_1));
+
+        TestEvent expectedEvent = new TestEvent(EventId.first(), AGGREGATE_ID, PAYLOAD_1);
+        assertThat(eventStore.getEventsOfAggregate(AGGREGATE_ID).getEvents())
+            .containsOnly(expectedEvent);
+    }
+
+    @Test
+    default void dispatchShouldCallSubscriberForSubsequentCommands(EventStore eventStore) {
+        DataCollectorSubscriber subscriber = new DataCollectorSubscriber();
+        EventSourcingSystem eventSourcingSystem = new EventSourcingSystem(
+            ImmutableSet.of(simpleDispatcher(eventStore)),
+            ImmutableSet.of(subscriber),
+            eventStore);
+
+        eventSourcingSystem.dispatch(new MyCommand(PAYLOAD_1));
+        eventSourcingSystem.dispatch(new MyCommand(PAYLOAD_2));
+
+        assertThat(subscriber.getData()).containsExactly(PAYLOAD_1, PAYLOAD_2);
+    }
+
+    @Test
+    default void dispatchShouldStoreEventsForSubsequentCommands(EventStore eventStore) {
+        DataCollectorSubscriber subscriber = new DataCollectorSubscriber();
+        EventSourcingSystem eventSourcingSystem = new EventSourcingSystem(
+            ImmutableSet.of(simpleDispatcher(eventStore)),
+            ImmutableSet.of(subscriber),
+            eventStore);
+
+        eventSourcingSystem.dispatch(new MyCommand(PAYLOAD_1));
+        eventSourcingSystem.dispatch(new MyCommand(PAYLOAD_2));
+
+        TestEvent expectedEvent1 = new TestEvent(EventId.first(), AGGREGATE_ID, PAYLOAD_1);
+        TestEvent expectedEvent2 = new TestEvent(expectedEvent1.eventId().next(), AGGREGATE_ID, PAYLOAD_2);
+        assertThat(eventStore.getEventsOfAggregate(AGGREGATE_ID).getEvents())
+            .containsOnly(expectedEvent1, expectedEvent2);
+    }
+
+    @Test
+    default void dispatcherShouldBeAbleToReturnSeveralEvents(EventStore eventStore) {
+        DataCollectorSubscriber subscriber = new DataCollectorSubscriber();
+        EventSourcingSystem eventSourcingSystem = new EventSourcingSystem(
+            ImmutableSet.of(wordCuttingDispatcher(eventStore)),
+            ImmutableSet.of(subscriber),
+            eventStore);
+
+        eventSourcingSystem.dispatch(new MyCommand("This is a test"));
+
+        assertThat(subscriber.getData()).containsExactly("This", "is", "a", "test");
+    }
+
+    @Test
+    default void unknownCommandsShouldBeIgnored(EventStore eventStore) {
```

Comment:
It should not, actually. Such a system is not supposed to exists

Reply:
Ok, then let's throw!